sdk/tables/azure-data-tables/azure/data/tables/_models.py

```diff
@@ -473,31 +448,26 @@ def service_properties_deserialize(generated: GenTableServiceProperties) -> Dict
         ),
         "hour_metrics": TableMetrics._from_generated(generated.hour_metrics),  # pylint: disable=protected-access
         "minute_metrics": TableMetrics._from_generated(generated.minute_metrics),  # pylint: disable=protected-access
-        "cors": [
-            TableCorsRule._from_generated(cors)  # pylint: disable=protected-access
-            for cors in generated.cors  # type: ignore
-        ],
+        "cors": [TableCorsRule._from_generated(cors) for cors in generated.cors]  # pylint: disable=protected-access
+        if generated.cors
+        else generated.cors,
     }
 
 
 class TableItem:
-    """
-    Represents an Azure TableItem.
+    """Represents an Azure TableItem.
     Returned by TableServiceClient.list_tables and TableServiceClient.query_tables.
-
-    :ivar str name: The name of the table.
     """
 
+    name: str
+    """The name of the table."""
+
     def __init__(self, name: str) -> None:
-        """
-        :param str name: Name of the Table
-        """
         self.name = name
 
-    # TODO: TableQueryResponse is not the correct type
     @classmethod
-    def _from_generated(cls, generated: TableQueryResponse, **kwargs) -> "TableItem":  # pylint: disable=unused-argument
-        return cls(generated.table_name)  # type: ignore
+    def _from_generated(cls, generated: TableResponseProperties) -> "TableItem":
+        return cls(generated.table_name)  # type: ignore[arg-type] # table_name in generated model is not required.
```

Comment:
We can probably avoid this warning by falling back to table_name that was passed into `create_table`.
I believe for `list_tables`/`query_tables`, table_name should be returned as required.

In fact - we could simply remove the `_from_generated` method completely. And just pass `name` in from the method that constructs these.

Reply:
Yup, have removed this `_from_generated`.